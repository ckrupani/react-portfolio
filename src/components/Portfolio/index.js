import React, { useState } from "react";
import Project from "../Project";

function Portfolio({ title }) {
    const [projects] = useState([
        {
            name: "Food Store",
            description: "It is a webportal to buy food.",
            appLink: "",
            repoLink: "",
        },
        {
            name: "Tours and Travel",
            description: "It is a webportal to organize tours.",
            appLink: "",
            repoLink: "",
        },
        {
            name: "Time Store",
            description: "It is a webportal to buy timber.",
            appLink: "",
            repoLink: "",
        },
        {
            name: "AirBnB",
            description: "It is a webportal to book homes.",
            appLink: "",
            repoLink: "",
        },
        {
            name: "Landscaping",
            description:
                "It is a webportal to organize your landscaping needs.",
            appLink: "",
            repoLink: "",
        },
    ]);

    return (
        <div>
            <h1>{title}</h1>
            <div className="flex-row">
                {projects.map((project, i) => (
                    <Project
                        key={project.name}
                        projectInfo={project}
                        imageId={i}
                    />
                ))}
            </div>
        </div>
    );
}

export default Portfolio;
