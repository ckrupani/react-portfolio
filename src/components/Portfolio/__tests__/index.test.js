import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Portfolio from "..";

afterEach(cleanup);

describe("Portfolio component", () => {
    it("renders", () => {
        render(<Portfolio title={"Portfolio"} />);
    });

    it("matches snapshot DOM node structure", () => {
        const { asFragment } = render(<Portfolio title={"Portfolio"} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
