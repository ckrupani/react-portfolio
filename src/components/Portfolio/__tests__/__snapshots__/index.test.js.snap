// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Portfolio
    </h1>
    <div
      class="flex-row"
    >
      <div
        class="project-wrap"
      >
        <img
          alt="Food Store"
          class="img-thumbnail mx-1"
          src="0.jpg"
        />
        <div
          class="project-description"
        >
          <h3>
            <a
              href=""
              rel="noreferrer"
              target="_blank"
            >
              Food Store
            </a>
          </h3>
          <p>
            It is a webportal to buy food.
          </p>
          <a
            href=""
            rel="noreferrer"
            target="_blank"
          >
            Github
          </a>
        </div>
      </div>
      <div
        class="project-wrap"
      >
        <img
          alt="Tours and Travel"
          class="img-thumbnail mx-1"
          src="1.jpg"
        />
        <div
          class="project-description"
        >
          <h3>
            <a
              href=""
              rel="noreferrer"
              target="_blank"
            >
              Tours and Travel
            </a>
          </h3>
          <p>
            It is a webportal to organize tours.
          </p>
          <a
            href=""
            rel="noreferrer"
            target="_blank"
          >
            Github
          </a>
        </div>
      </div>
      <div
        class="project-wrap"
      >
        <img
          alt="Time Store"
          class="img-thumbnail mx-1"
          src="2.jpg"
        />
        <div
          class="project-description"
        >
          <h3>
            <a
              href=""
              rel="noreferrer"
              target="_blank"
            >
              Time Store
            </a>
          </h3>
          <p>
            It is a webportal to buy timber.
          </p>
          <a
            href=""
            rel="noreferrer"
            target="_blank"
          >
            Github
          </a>
        </div>
      </div>
      <div
        class="project-wrap"
      >
        <img
          alt="AirBnB"
          class="img-thumbnail mx-1"
          src="3.jpg"
        />
        <div
          class="project-description"
        >
          <h3>
            <a
              href=""
              rel="noreferrer"
              target="_blank"
            >
              AirBnB
            </a>
          </h3>
          <p>
            It is a webportal to book homes.
          </p>
          <a
            href=""
            rel="noreferrer"
            target="_blank"
          >
            Github
          </a>
        </div>
      </div>
      <div
        class="project-wrap"
      >
        <img
          alt="Landscaping"
          class="img-thumbnail mx-1"
          src="4.jpg"
        />
        <div
          class="project-description"
        >
          <h3>
            <a
              href=""
              rel="noreferrer"
              target="_blank"
            >
              Landscaping
            </a>
          </h3>
          <p>
            It is a webportal to organize your landscaping needs.
          </p>
          <a
            href=""
            rel="noreferrer"
            target="_blank"
          >
            Github
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
